File: Problems/No.1823_Find_the_Winner_of_the_Cirlular_Game/find_the_winner_of_the_circular_game.cpp
class Solution {
public:
    int findTheWinner(int n, int k) {
        
        // parameter
        std::queue<int> friends;

        // start computing
        for (int i=1; i<=n; i++) friends.push(i);
        while (friends.size() > 1) {

            for (int cnt=1; cnt<k; cnt++) {

                friends.push(friends.front());
                friends.pop();

            }
            friends.pop();

        }

        return friends.front();

    }
};

File: Problems/No.1823_Find_the_Winner_of_the_Cirlular_Game/find_the_winner_of_the_circular_game.py
class Solution(object):
    def findTheWinner(self, n, k):
        """
        :type n: int
        :type k: int
        :rtype: int
        """

        # parameter
        friends = range(1, n+1)

        # start computing
        while (len(friends) > 1):
            for i in range(1, k):
                friends.append(friends.pop(0))
            friends.pop(0)

        return friends[0]
